data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -100,6 +108,46 @@
       final BLSSignature randao,
       final Optional<Bytes32> graffiti,
       final boolean isBlinded) {
+    checkBlockProducingParameters(slot, randao);
+    return validatorApiChannel.createUnsignedBlock(slot, randao, graffiti, isBlinded);
+  }
+
+  public SafeFuture<Optional<BlockContainerAndMetaData>> produceBlock(
+      final UInt64 slot, final BLSSignature randao, final Optional<Bytes32> graffiti) {
+    checkBlockProducingParameters(slot, randao);
+    return validatorApiChannel
+        .createUnsignedBlock(slot, randao, graffiti)
+        .thenCombine(retrieveExecutionPayloadValue(slot), this::lookUpData);
+  }
+
+  private Optional<BlockContainerAndMetaData> lookUpData(
+      Optional<BlockContainer> maybeBlockContainer, UInt256 executionPayloadValue) {
+    if (maybeBlockContainer.isEmpty()) {
+      return Optional.empty();
+    } else {
+      return Optional.of(
+          new BlockContainerAndMetaData(
+              maybeBlockContainer.get(),
+              spec.atSlot(maybeBlockContainer.get().getSlot()).getMilestone(),
+              false,
+              false,
```

Comment:
I'm not sure it's not canonical. It looks like a bad sign when reusing ObjectAndMetadata that we actually need only 2 of its 5 fields. Maybe it worth setting up a separate class for this (record?)

Reply:
Done. I refactored the `BlockContainerAndMetadata`, created a record and added the ssz serialisation